host/app/services/local-realm.ts

```diff
@@ -202,6 +202,13 @@ export default class LocalRealm extends Service {
     while (registration.active?.state !== 'activated') {
       await timeout(10);
     }
+    navigator.serviceWorker.oncontrollerchange = () => {
+      console.log('worker changed');
+      if ('worker' in this.state) {
+        // update worker reference to the new one
+        this.state.worker = navigator.serviceWorker.controller!;
```

Comment:
We discussed changing this so it resets the state to starting-up and calls maybeSetup() again.